### ISSUE TITLE
Toolkit is not compatible with Node.js Environment

### ISSUE THREAD

**Issue by jboulmal** [reTHINK-project/dev-hyperty-toolkit#32]
The bundle process for server side Hyperties to be served by the toolkit is not compatible with Runtime Node. 
Mainly, if the NodeHyperty requires some dependencies, we got the following error on Runtime Node:
``` 
return binding.fstat(fd);

ERROR: { Error: EBADF: bad file descriptor, fstat
    at Error (native)
    at Object.fs.fstatSync (fs.js:971:18)
    at tryStatSync (fs.js:456:13)
    at fs.readFileSync (fs.js:504:12)
    at Server.serveClient (eval at <anonymous> (null:756:2), <anonymous>:97:20)
    at new Server (eval at <anonymous> (null:756:2), <anonymous>:50:8)
    at Function.Server (eval at <anonymous> (null:756:2), <anonymous>:42:41)
    at eval (eval at <anonymous> (null:48:2), <anonymous>:70:35)
    at Object.<anonymous> (null:48:2)
    at __webpack_require__ (null:21:30) errno: -9, code: 'EBADF', syscall: 'fstat' }

**Comment by pchainho** [comment]
Dependencies should be provided in the hyperty distribution file and if that is not possible they should be added as nodejs runtime capabilities and added as constraints in the hyperty descriptor

**Comment by jboulmal** [comment]
We have solved this by adding to [transpile.js](https://github.com/reTHINK-project/dev-hyperty-toolkit/blob/develop/gulp/transpile.js) :

`var nodeExternals = require('webpack-node-externals');`
Besides, I have added ` externals: [nodeExternals()],` after [target: 'node',](https://github.com/reTHINK-project/dev-hyperty-toolkit/blob/develop/gulp/transpile.js#L132) and replace [ { exclude:.............](https://github.com/reTHINK-project/dev-hyperty-toolkit/blob/develop/gulp/transpile.js#L143) by ` { exclude: 'node_modules', test: /\.js$/, loader: "babel-loader" }`

As commented above by @pchainho. The idea is to have dependencies installed in the runtime-nodejs that is the final target/platform of the app, where it will be executed.

**Comment by pchainho** [comment]
The toolkit should also support IdpProxy encoding for nodejs

**Comment by pchainho** [comment]
The toolkit is currently compliant with NodeJS. 

A better management of dependencies will be addressed in #39
